layer/contracts/aws.py

```diff
@@ -0,0 +1,14 @@
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class AWSCredentials:
+    access_key_id: str
+    secret_access_key: str
+    session_token: str
```

Comment:
where do we put the types we officially support for sdk users?

SDK internals vs SDK public interfaces? are all contracts supposed to be supported? or they're simply plain python code ~ core domain inside the SDK between our APIs and User interfaces?

Reply:
haven't thought through that part yet :) but I do like the idea of having separate classes for publicly supported objects vs internal domain objects. maybe a task for a future PR.